components/chat_widget/src/components/ocs-chat/ocs-chat.tsx

```diff
@@ -309,13 +337,79 @@ export class OcsChat {
     }
   }
 
+  private async uploadFiles(): Promise<number[]> {
+    if (this.selectedFiles.length === 0 || !this.sessionId || !this.allowAttachments) {
+      return [];
+    }
+
+    this.isUploadingFiles = true;
+    const uploadedIds: number[] = [];
+
+    try {
+      const formData = new FormData();
+
+      // Add all files to form data
+      for (const selectedFile of this.selectedFiles) {
+        if (!selectedFile.error && !selectedFile.uploaded) {
+          formData.append('files', selectedFile.file);
+        } else if (selectedFile.uploaded) {
+          uploadedIds.push(selectedFile.uploaded.id);
+        }
+      }
+
+      // Add user ID and name to the form data
+      const userId = this.getOrGenerateUserId();
+      formData.append('participant_remote_id', userId);
+      if (this.userName) {
+        formData.append('participant_name', this.userName);
+      }
+
+      // Only upload if there are new files
+      if (formData.has('files')) {
+        const response = await fetch(`${this.getApiBaseUrl()}/api/chat/${this.sessionId}/upload/`, {
+          method: 'POST',
+          body: formData,
+        });
+
+        if (!response.ok) {
+          const errorData = await response.json();
+          throw new Error(errorData.error || 'Failed to upload files');
```

Comment:
This should rather update the selecteFiles to add the `error` field so that the UI displays the error instead of closing the whole chat window.